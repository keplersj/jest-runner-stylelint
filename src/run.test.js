const path = require("path");
const run = require("./run");

// Remove undeterministic data from test reports
expect.addSnapshotSerializer({
  print: (value, serialize) => {
    delete value.perfStats;
    delete value.testFilePath;
    value.testResults.forEach((result) => {
      delete result.duration;
    });
    return serialize(value);
  },
  test: (value) =>
    value && value.perfStats && value.testFilePath && value.testResults,
});

describe("jest-runner-stylelint", () => {
  describe("failing fixture", () => {
    it("matches snapshot", () =>
      run({
        testPath: path.join(__dirname, "__fixtures__", "bad.css"),
        config: {},
        globalConfig: {},
        extraOptions: {},
      }).then((result) => expect(result).toMatchSnapshot()));
  });

  describe("passing fixture", () => {
    it("matches snapshot", () =>
      run({
        testPath: path.join(__dirname, "__fixtures__", "good.css"),
        config: {},
        globalConfig: {},
        extraOptions: {},
      }).then((result) => expect(result).toMatchSnapshot()));
  });
});
