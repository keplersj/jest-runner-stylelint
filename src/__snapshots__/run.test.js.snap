// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-runner-stylelint failing fixture matches snapshot 1`] = `
Object {
  "console": null,
  "failureMessage": "
[4msrc/__fixtures__/bad.css[24m
 [2m2:11[22m  [31m[31mâœ–[31m[39m  Unexpected unit   [2mlength-zero-no-unit[22m
 [2m3:12[22m  [31m[31mâœ–[31m[39m  Unexpected unit   [2mlength-zero-no-unit[22m

",
  "numFailingTests": 1,
  "numPassingTests": 0,
  "numPendingTests": 0,
  "numTodoTests": 0,
  "skipped": undefined,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "unmatched": 0,
    "updated": 0,
  },
  "sourceMaps": Object {},
  "testExecError": null,
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "failureMessages": Array [
        "
[4msrc/__fixtures__/bad.css[24m
 [2m2:11[22m  [31m[31mâœ–[31m[39m  Unexpected unit   [2mlength-zero-no-unit[22m
 [2m3:12[22m  [31m[31mâœ–[31m[39m  Unexpected unit   [2mlength-zero-no-unit[22m

",
      ],
      "fullName": undefined,
      "numPassingAsserts": 1,
      "status": "failed",
      "title": "",
    },
  ],
}
`;

exports[`jest-runner-stylelint passing fixture matches snapshot 1`] = `
Object {
  "console": null,
  "failureMessage": undefined,
  "numFailingTests": 0,
  "numPassingTests": 1,
  "numPendingTests": 0,
  "numTodoTests": 0,
  "skipped": undefined,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "unmatched": 0,
    "updated": 0,
  },
  "sourceMaps": Object {},
  "testExecError": null,
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "failureMessages": Array [
        undefined,
      ],
      "fullName": undefined,
      "numPassingAsserts": 0,
      "status": "passed",
      "title": "",
    },
  ],
}
`;
